File: Src/Visual Basic .NET/FTP-related/ClientFtp/LocalListViewItem.vb
Imports System.Windows.Forms
Imports Xceed.FileSystem

Public Class LocalListViewItem
  Inherits ListViewItem

  Public Sub New(ByVal item As FileSystemItem)
    MyBase.New(item.Name)

    If item Is Nothing Then
      Throw New ArgumentNullException("item")
    End If

    If TypeOf item Is AbstractFile Then
      Dim file As AbstractFile = item

      Me.SubItems.Add(RemoteFolderTreeNode.FormatSize(file.Size))
      Me.ImageIndex = CInt(FtpItemIconEnum.File)
    Else
      Me.SubItems.Add("")
      Me.ImageIndex = CInt(FtpItemIconEnum.ClosedFolder)
    End If

    Me.SubItems.Add(item.LastWriteDateTime.ToString())

    m_item = item
  End Sub


  Public ReadOnly Property Item() As FileSystemItem
    Get
      Return m_item
    End Get
  End Property


  Private m_item As FileSystemItem = Nothing
End Class

File: Src/DetailedSamples/CSharp/FTP-related/ClientFtp/LocalListViewItem.cs
using System;
using System.Windows.Forms;
using Xceed.FileSystem;

namespace ClientFtp
{
	public class LocalListViewItem : ListViewItem
	{
		public LocalListViewItem( FileSystemItem item )
      : base( item.Name )
		{
      if( item == null )
        throw new ArgumentNullException( "item" );

      AbstractFile file = item as AbstractFile;

      // We only show the size for a File.
      if( file != null )
      {
        this.SubItems.Add( RemoteFolderTreeNode.FormatSize( file.Size ) );
        this.ImageIndex = ( int ) FtpItemIconEnum.File;
      }
      else
      {
        this.SubItems.Add( "" );
        this.ImageIndex = ( int ) FtpItemIconEnum.ClosedFolder;
      }

      this.SubItems.Add( item.LastWriteDateTime.ToString() );

      m_item = item;
    }

    public FileSystemItem Item
    {
      get { return m_item; }
    }

    private FileSystemItem m_item = null;
	}
}
